src/dafny/NonlinearArithmetic/NonlinearArithmetic.md

```diff
@@ -0,0 +1,11 @@
+
+## The `NonlinearArith` module {#sec-nonlinear-arith}
+
+Non-linear math is generally undecidable; hence Dafny (and Z3) rely on heuristics to prove such properties.
```

Comment:
Can we briefly define non-linear or link somewhere that does?

Reply:
Brief sentence included here.